### PR TITLE
Server working with 'winston' and 'sendgrid'

### PR DESCRIPTION
**Pull request by runtilov**

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -1,6 +0,0 @@
-{
```
Comment (alxolr) [inline .eslintrc.json]: Why did you remove the .eslintrc.json file?

File: config/default.js [hunk 2]
```diff
@@ -1,4 +1,5 @@
 module.exports = {
-  mysql: process.env.MYSQL_URI || 'mysql://localhost:3306/moodle',
-  facebookApiKey: process.env.FACEBOOK_API_KEY,
+    mysql: process.env.MYSQL_URI || 'mysql://root:parola@192.168.99.100:3306/moodle',
```
Comment (alxolr) [inline config/default.js]: Try to not use credentials in the environment vars. `root:parola`

File: index.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+const sgMail = require('@sendgrid/mail');
```
Comment (alxolr) [inline index.js]: Please uniformize the code style to be `2` spaces instead of `4` and be consistent through the entire project.

File: config/default.js [hunk 4]
```diff
@@ -1,4 +1,5 @@
...
+    sendgrid_api_key: process.env.SENDGRID_API_KEY || 'test',
```
Comment (alxolr) [inline config/default.js]: Uniformise the variable name to be consistent with the `cameCase` style, `sendgridApiKey`

File: index.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+const connFactory = require('./src//services/connectors/connector.factory');
+
+connFactory()
+    .connect()
+    .then((connection) => {
+        const app = require('./src/app')(connection);
+
+        const port = process.env.PORT || 3000;
```
Comment (alxolr) [inline index.js]: It is a good practice to set all the environment variables in the config file, extract port in the config file

File: package.json [hunk 6]
```diff
@@ -1,19 +1,28 @@
...
+    "app-root-path": "^2.1.0",
+    "body-parser": "^1.18.3",
     "config": "^1.30.0",
     "express": "^4.16.3",
-    "winston": "^3.0.0"
+    "morgan": "^1.9.0",
+    "mysql2": "^1.5.3",
+    "sequelize": "^4.38.0",
+    "winston": "^3.0.0",
+    "winston-mail": "^1.5.1"
```
Comment (alxolr) [inline package.json]: It seems you are not using the `winston-mail` package please tide up the unused packages.

File: src/services/winston/sendgrid.transport.js [hunk 7]
```diff
@@ -0,0 +1,25 @@
+const winston = require('winston');
+
+module.exports = class SendGridTransport extends winston.Transport {
+    constructor(options) {
+        super(options);
+        this.name = 'SendGridLogger';
+        this.level = (options && options.level) || 'error';
+        this.server = options && options.server;
```
Comment (alxolr) [inline src/services/winston/sendgrid.transport.js]: Server is an abigous name for the `mailServer` please rename it.

File: src/services/winston/sendgrid.transport.js [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+    }
+
+    log(level, msg, meta, callback) {
+        const message = {
+            to: this.to,
+            from: this.from,
+            subject: this.subject,
+            text: `Metadata: ${meta}\n Info: ${this.text}`,
+        };
+        this.server.send(message);
```
Comment (alxolr) [inline src/services/winston/sendgrid.transport.js]: Sendgrid `send` method is an asynchronous one, you can attach a callback in the following form, 
```
this.server.send(message, () => {}); 
// or in our case just use the log callback provided by the winston.transport
this.server.send(message, callback)
```

File: src/services/winston/winston.js [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+    },
+    console: {
+        level: 'debug',
+        handleExceptions: true,
+        json: false,
+        colorize: true,
+    },
+    sendgrid: {
+        level: 'error',
+        to: 'runtilov@pentalog.com',
```
Comment (alxolr) [inline src/services/winston/winston.js]: Please do not hardcode emails, or other credentials in the code, exctract it in a separate config file or environment variable, and make sure, you can provide a list of emails for the people responsible.

File: src/services/winston/sendgrid.transport.js [hunk 10]
```diff
@@ -0,0 +1,25 @@
+const winston = require('winston');
+
+module.exports = class SendGridTransport extends winston.Transport {
+    constructor(options) {
+        super(options);
+        this.name = 'SendGridLogger';
+        this.level = (options && options.level) || 'error';
+        this.server = options && options.server;
+        this.to = (options && options.to) || 'runtilov@pentalog.com';
```
Comment (alxolr) [inline src/services/winston/sendgrid.transport.js]: The following part
```
this.to
this.from
this.subject
this.text
```
Can be extracted into a `Message` class and reused throughout the project.

File: src/services/winston/winston.js [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+
+sgMail.setApiKey(config.sendgrid_api_key);
+
+const opt = {
+    file: {
+        level: 'info',
+        filename: `${appRoot}/logs/app.log`,
+        handleExceptions: true,
+        json: true,
+        maxsize: 5242880, // 5MB
```
Comment (alxolr) [inline src/services/winston/winston.js]: Sometimes when you are working with the size it is a good practice also to write it in the following manner `5 * 1024 * 1024` And it's clear that we have 5MB or you can add also a comment like you added also is a good practice.

File: .eslintrc.json (hunk 1 above)
Comment (runtilov) [reply .eslintrc.json]: I removed it so that others can use their own styles

Comment (runtilov) [reply .eslintrc.json]: I expected to keep my version of it, but apparently I somehow deleted my '.eslintrc.json' as well, my bad.

File: .gitignore [hunk 12]
```diff
@@ -1 +1,5 @@
-node_modules/
\ No newline at end of file
+node_modules/
+.vscode/
+app.yml
```
Comment (alxolr) [inline .gitignore]: What are the yml files.

File: src/helpers/util.js [hunk 13]
```diff
@@ -1,7 +1,26 @@
-module.exports = {
-  add,
-};
-
 function add(a, b) {
   return a + b;
 }
+
+// class Message {
```
Comment (alxolr) [inline src/helpers/util.js]: There should not be commented cod

File: test/src/helpers/util.test.js [hunk 14]
```diff
@@ -5,6 +5,14 @@ const util = require('../../../src/helpers/util');
 
 describe('util', () => {
   it('should sum 2 numbers', () => {
-    assert.equal(util.add(2, 5), 7);
+    assert.strictEqual(util.add(2, 5), 7);
   });
```
Comment (alxolr) [inline test/src/helpers/util.test.js]: There is commented code.
